File: PokemonGoRaidBot/src/api/mc.raid.details.ts
import { McRaidDescription } from "./mc.raid.description";
import { McUserParticipation } from "./mc.user.particiation";
import { McTelegramUser } from "./mc.telegram.user";

export class McRaidDetails {
    public UniqueID: string;
    public PublicID: string;
    public Raid: McRaidDescription;
    public IsPublished: boolean;
    public LastRefresh: string;
    public LastModificationTime: string;
    public Participants: McUserParticipation;
    public Rejected: McTelegramUser[];
    public Done: McTelegramUser[];
    public Maybe: McTelegramUser[];
    //
    public NumberOfParticipants: number;
}

File: PokemonGoRaidBot/Backend/API/McRaidDetails.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using RaidBot.Backend.Bot.PokemonRaidBot.Entities;
using System;
using System.Collections.Generic;

namespace RaidBot.Backend.API
{
    [JsonObject(MemberSerialization.OptOut)]
    public class McRaidDetails
    {
        public Guid UniqueID { get; set; }

        public string PublicID { get; set; }

        public McRaidDescription Raid { get; set; }

        public bool IsPublished { get; set; }

        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTime LastRefresh { get; set; }

        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTime LastModificationTime { get; set; }

        public Dictionary<Team, List<McUserParticipation>> Participants { get; set; }

        public List<McTelegramUser> Rejected { get; set; }

        public List<McTelegramUser> Done { get; set; }

        public List<McTelegramUser> Maybe { get; set; }
    }
}
